### DISCUSSION TITLE
Reference block time may be invalid

### DISCUSSION THREAD

**Discussion by matthiasgeihs** [hyperledger-labs/go-perun#68] [General]
Location: funder.go, timeout.go

When we fund, we wait for blocktime + challengeDuration before we cancel the context. In some situations this can cause unwanted behavior. For example, when the block tick time is larger than the challenge duration, the BlockTimeout is immediately canceled with the next block, even it the next block arrives within the expected time frame.

**Comment by ggwpez** [comment]
The challenge duration should *always* be at least 3 block-times. Otherwise the user risks losing funds.  
This looks more like incorrect usage to me.

**Comment by matthiasgeihs** [comment]
@ggwpez  After thinking about it, I agree.

However, the restriction on the challenge duration is not self-explaining and also not checked by the framework. Do we document it somewhere?
